src/components/EditorSideBar/Blueprints.tsx

```diff
@@ -7,12 +7,22 @@ import { setActiveCourt } from "@/store/reducer/courtSpecDataSlice";
 import { mockTileData } from "../MockData/MockTileData";
 import { useStoreSelector } from "@/store/hooks";
 import { resetAll } from "@/store/reducer/canvasControlSlice";
+import { ICourtData } from "@/interfaces/design";
 
-const Blueprints: React.FC = () => {
+interface Props {
+  fetchedCourtsData: ICourtData[];
+}
+const Blueprints = ({ fetchedCourtsData }: Props) => {
+  console.log(fetchedCourtsData);
   const dispatch = useDispatch();
   const [activateCourt, setActivateCourt] = useState<string>("");
   const { courtsData } = useStoreSelector((state) => state.courtSpecData);
 
+  const filteredCourtList = courtList.filter((court) =>
+    fetchedCourtsData
+      ?.filter((item: ICourtData) => item.isHidden === false)
+      .some((item: ICourtData) => item.name === court.courtSizeName)
+  );
```

Comment:
I believe this logic can be simplified. Give it a second try.

Reply:
Done~thanks